scripts/run_presubmit_checks_test.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/run_presubmit_checks.py."""
+
+from __future__ import annotations
+
+import builtins
+import os
+import subprocess
+
+from core.tests import test_utils
+
+from scripts import run_backend_tests
+from scripts import run_frontend_tests
+from scripts import run_presubmit_checks
+from scripts.linters import pre_commit_linter
+
+
+class RunPresubmitChecksTests(test_utils.GenericTestBase):
+    """Unit tests for scripts/run_presubmit_checks.py."""
+
+    def setUp(self) -> None:
+        super(RunPresubmitChecksTests, self).setUp()
+        self.print_arr: list[str] = []
+        def mock_print(msg: str) -> None:
+            self.print_arr.append(msg)
+        self.print_swap = self.swap(builtins, 'print', mock_print)
+
+        current_dir = os.path.abspath(os.getcwd())
+        self.changed_frontend_file = os.path.join(
+            current_dir, 'core', 'templates', 'tesFile.ts')
+        self.current_branch = 'test-branch'
+        self.cmd_to_check_current_branch = [
+            'git', 'rev-parse', '--abbrev-ref', 'HEAD']
+        self.cmd_to_match_current_branch_with_remote = [
+        'git', 'ls-remote', '--heads', 'origin', self.current_branch, '|',
+        'wc', '-l']
+
+        self.scripts_called = {
+            'run_frontend_tests': False,
+            'run_backend_tests': False,
+            'pre_commit_linter': False
+        }
+        def mock_frontend_tests(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['run_frontend_tests'] = True
+        def mock_backend_tests(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['run_backend_tests'] = True
+        def mock_pre_commit_linter(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['pre_commit_linter'] = True
+
+        self.swap_frontend_tests = self.swap(
+            run_frontend_tests, 'main', mock_frontend_tests)
+        self.swap_backend_tests = self.swap(
+            run_backend_tests, 'main', mock_backend_tests)
+        self.swap_pre_commit_linter = self.swap(
+            pre_commit_linter, 'main', mock_pre_commit_linter)
```

Comment:
Instead of using `self.scripts_called`, you could just use `self.swap_with_checks`. That would also let you specify the expected args each function is called with to make the tests more thorough

Reply:
Done!